docs/log.rst

```diff
@@ -1,106 +1,114 @@
-   end of the file.
+    If a log file already exists when the programme starts, or if this function
+    is called multiple times, it will check the labels already defined in the
+    log file.
+    If this function call contains any new labels not already present, it will
+    generate a new header row with the additional columns.
 
-* **value**: Positional arguments used to generate the log headers,
-  which go on the first line of the CSV file. For example, ``set_labels("A","B","C")``
-  will create three column headers titled ``A``, ``B`` and ``C`` in that order. 
-* **timestamp**: Select the timestamp unit that will be automatically
-  added as the first column in every row. Timestamp values can be one of
-  ``MILLISECONDS``, ``SECONDS``, ``MINUTES``, ``HOURS``, ``DAYS`` or ``None`` to
-  disable the timestamp.
+    By default the first column contains a time stamp for each row. The time
+    unit can be selected via the ``timestamp`` argument, e.g.
+    ``log.set_labels("temp", timestamp=log.MINUTES)``
 
-.. py:function:: set_mirroring(value)
+    :param \*labels: Any number of positional arguments, each corresponding to
+        an entry in the log header.
+    :param timestamp: Select the timestamp unit that will be automatically
+        added as the first column in every row. Timestamp values can be one of
+        ``log.MILLISECONDS``, ``log.SECONDS``, ``log.MINUTES``, ``log.HOURS``,
+        ``log.DAYS`` or ``None`` to disable the timestamp. The default value
+        is ``log.SECONDS``.
 
-   Mirrors the datalogging activity to the serial output.
-   Serial mirroring is off by default. 
+.. py:function:: set_mirroring(serial)
 
-* **value**: Boolean. ``True`` will enable mirroring data to the serial output.
+    Configure mirroring of the data logging activity to the serial output.
+
+    Serial mirroring is disabled by default. When enabled, it will print to
+    serial each row logged into the log file.
+
+    :param serial: ``True`` enables mirroring data to the serial output.
 
 .. py:function:: delete(full=False)
 
-   Deletes the contents of the log, including headers.
-   To add the log headers the ``set_labels`` function has to be called again
-   after this.
+    Delete the contents of the log, including headers.
+
+    To add the log headers again the ``set_labels`` function should to be
+    called after this function.
 
-* **full**: Selects a "full" erase format that removes the data from the
-  flash storage. If set to ``False`` it uses a "fast" method,
-  which invalidates the data instead of performing a slower 
-  full erase.
+    There are two erase modes, "full" completely removes the data from the
+    physical storage, and the "fast" method only invalidates the data.
 
-.. py:function:: add({key:value})
-                 add(key=value)
-   
-    There are two ways to add a data row into the log:
+    :param full: ``True`` selects a "full erase" and ``False`` selects the
```

Comment:
Rather than having quote marks for "full erase", just quote the mode names "full" and "fast" as it will be more consistent in help docs

```suggestion
    :param full: ``True`` selects a "full" erase and ``False`` selects the "fast" erase method.
```